obspy/clients/fdsn/client.py

```diff
@@ -1078,7 +1078,10 @@ def _get_bulk_string(self, bulk, arguments):
                 msg = ("Unrecognized input for 'bulk' argument. Please "
                        "contact developers if you think this is a bug.")
                 raise NotImplementedError(msg)
-        return bulk.encode('ASCII')
+        if PY2:
+            return bulk
+        else:
+            return bulk.encode('ASCII')
```

Comment:
I can't say for sure without digging through the code and investigating which test fail you're fixing here (and what that test is doing), the main question is where this return is getting used..

But, if you're running into a situation that you have to do an encoding or not, depending on Py2/3 maybe the problem is further up.. it's best practice to always do explicit de/encoding on I/O.